pcdsdevices/gon.py

```diff
@@ -236,23 +266,290 @@ class Kappa(BaseInterface, Device):
+            phi = -self.phi.position
+
+        kappa_ang = self.kappa_ang * np.pi / 180
+
+        delta = np.arctan(np.tan(kappa * np.pi / 180 / 2.0) *
+                          np.cos(kappa_ang))
+        e_eta = -eta * np.pi / 180 - delta
+        e_chi = 2.0 * np.arcsin(np.sin(kappa * np.pi / 180 / 2.0) *
+                                np.sin(kappa_ang))
+        e_phi = phi * np.pi / 180 - delta
+        e_eta = e_eta * 180 / np.pi
+        e_chi = e_chi * 180 / np.pi
+        e_phi = e_phi * 180 / np.pi
+        return e_eta, e_chi, e_phi
+
+    def e_to_k(self, e_eta=None, e_chi=None, e_phi=None):
+        """
+        Convert from spherical coordinates to the native kappa coordinates.
+
+        If a parameter is left as None, use the live value.
+
+        Parameters
+        ----------
+        e_eta : number
+            e_eta pseudo motor's spherical coordinate
+        e_chi : number
+            e_chi pseudo motor's spherical coordinate
+        e_phi : number
+            e_phi pseudo motor's spherical coordinate
+
+        Returns
+        -------
+        coordinates : tuple
+            Native kappa coordinates.
+        """
+        if not e_eta and e_eta != 0:
+            e_eta = self.e_eta_coord
+        if not e_chi and e_chi != 0:
+            e_chi = self.e_chi_coord
+        if not e_phi and e_phi != 0:
+            e_phi = self.e_phi_coord
+
+        kappa_ang = self.kappa_ang * np.pi / 180
+
+        delta = np.arcsin(-np.tan(e_chi * np.pi / 180 /
+                                  2.0) / np.tan(kappa_ang))
+        k_eta = -(e_eta * np.pi / 180 - delta)
+        k_kap = 2.0 * np.arcsin(np.sin(e_chi * np.pi /
+                                       180 / 2.0) / np.sin(kappa_ang))
+        k_phi = e_phi * np.pi / 180 - delta
+
+        k_eta = k_eta * 180 / np.pi
+        k_kap = k_kap * 180 / np.pi
+        k_phi = -k_phi * 180 / np.pi
+        return k_eta, k_kap, k_phi
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos):
+        """
+        Calculate a RealPosition from a given PseudoPosition.
+
+        Parameters
+        ----------
+        pseudo_pos : PseudoPosition
+            The pseudo position input.
+        Returns
+        -------
+        real_position : RealPosition
+            The real position output.
+        """
+        pseudo_pos = self.PseudoPosition(*pseudo_pos)
+        eta, kappa, phi = self.e_to_k(pseudo_pos.e_eta, pseudo_pos.e_chi,
+                                      pseudo_pos.e_phi)
+        return self.RealPosition(eta=eta, kappa=kappa, phi=phi)
+
+    @pseudo_position_argument
+    def move(self, position, wait=True, timeout=None, moved_cb=None):
+        """
+        Move to a specified position, optionally waiting for motion to
+        complete.
+
+        Checks for the motor step, and ask the user for confirmation if
+        movement step is greater than default one.
+        """
+        if self.check_motor_step(position.e_eta, position.e_chi,
```

Comment:
I hope i'm not doing anything crazy here but this seems to work.

Also because I added this now, I removed the other mv methods: `mv_e_eta`, `mv_e_chi` and `mv_e_phi` - I think we don't need them anymore?

Reply:
I agree the the other motor mv methods are not needed and this seems like the correct approach to me